src/Microsoft.AspNetCore.Mvc.Core/ApiExplorer/ApiConventionResult.cs

```diff
@@ -0,0 +1,195 @@
+            // inside the for loop.
+            return methodParameters.Length == conventionMethodParameters.Length;
+
+            ApiConventionNameMatchBehavior GetNameMatchBehavior(ICustomAttributeProvider attributeProvider)
+            {
+                var attributes = attributeProvider.GetCustomAttributes(inherit: false);
+                for (var i = 0; i < attributes.Length; i++)
+                {
+                    if (attributes[i] is ApiConventionNameMatchAttribute nameMatchAttribute)
+                    {
+                        return nameMatchAttribute.MatchBehavior;
+                    }
+                }
+
+                return ApiConventionNameMatchBehavior.Exact;
+            }
+
+            ApiConventionTypeMatchBehavior GetTypeMatchBehavior(ParameterInfo parameter)
+            {
+                var typeMatchAttribute = parameter.GetCustomAttribute<ApiConventionTypeMatchAttribute>(inherit: false);
+                return typeMatchAttribute?.MatchBehavior ?? ApiConventionTypeMatchBehavior.AssignableFrom;
+            }
+        }
+
+        internal static bool IsNameMatch(string name, string conventionName, ApiConventionNameMatchBehavior nameMatchBehavior)
+        {
+            switch (nameMatchBehavior)
+            {
+                case ApiConventionNameMatchBehavior.Any:
+                    return true;
+
+                case ApiConventionNameMatchBehavior.Exact:
+                    return string.Equals(name, conventionName, StringComparison.Ordinal);
+
+                case ApiConventionNameMatchBehavior.Prefix:
+                    return IsNameMatchPrefix();
+
+                case ApiConventionNameMatchBehavior.Suffix:
+                    return IsNameMatchSuffix();
+
+                default:
+                    return false;
+            }
+
+            bool IsNameMatchPrefix()
+            {
+                if (!name.StartsWith(conventionName, StringComparison.Ordinal))
+                {
+                    return false;
+                }
+
+                if (name.Length == conventionName.Length)
+                {
+                    return true;
+                }
+
+                return char.IsUpper(name[conventionName.Length]);
+            }
+
+            bool IsNameMatchSuffix()
+            {
+                // name = id, conventionName = id
+                if (string.Equals(name, conventionName, StringComparison.Ordinal))
+                {
+                    return true;
+                }
+
+                if (name.Length <= conventionName.Length)
+                {
+                    return false;
+                }
+
+                // name = personId, conventionName = id
+                var index = name.Length - conventionName.Length - 1;
+                if (!char.IsLower(name[index]))
+                {
+                    return false;
+                }
+
+                index++;
+                if (name[index] != char.ToUpper(conventionName[0]))
```

Comment:
Interesting, so we require the convention names to always be interpreted as if they are pascal cased?